libff/algebra/curves/profile/algebra_groups_profile.cpp

```diff
@@ -13,39 +13,79 @@
 
 using namespace libff;
 
+static const size_t ADD_NUM_DIFFERENT_ELEMENTS = 1024;
+static const size_t ADD_NUM_ELEMENTS = 1024*1024;
```

Comment:
I think I asked the exact them question in an old PR, but don't remember the answer. How are these taken? You picked arbitrary numbers for the profling here, right? Does that necessarily need to be a power of 2? (may be worth dropping a one liner comment)

Reply:
I've added some comments for these. Hope that's clearer now.

Reply:
I was mostly wondering how these values were taken, not how these constants were used (i.e. 1024 = 2^10 -> do we need a power of 2 here? or is that value arbitrarily chosen?)

Reply:
No, there is no power-of-two requirement.  The `NUM_DIFFERENT_ELEMENTS` is the number of random elements generated, so it must be chosen to give some meaningful variety, but not take too long to generate.  SImilarly, `NUM_ELEMENTS` is the number of actual additions performed in the profile, so it must be sufficiently large to get a good measurement, but not take an unreasonable amount of time.

So in some sense the choice is arbitrary, but I can't see much reason to choose anything too far from these values, unless running on some extremely different hardware.

Reply:
Got it, thanks.